grafana/grafana
# Prometheus: Run Explore both queries trough backend

**What this PR does / why we need it**:
Part of Prometheus migration to backend data source #37784

This PR focuses on making Explore + both queries (instant and range) run trough backend. Only if all targets are instant, range or both and non-exemplar queries, we will run them trough backend. This way, the PR and changes are smaller and it is easier to test and review PR.

**Testing:**
- Run any (Range | Instant | Both) query with Exemplars: on with opened Inspector -> Query tab. All queries should be run trough proxy.
- Toggle Exemplars: off and run Both query with opened Inspector -> Query tab. Query should be run trough backend.
- Toggle Exemplars: off and run Instant query ALERTS. Query should be run trough backend (the request url should be api/ds/query.
- Toggle Exemplars: off and run Range query ALERTS. Query should be run trough backend (the request url should be api/ds/query.
- Add second query and try to hide first query, then both queries.
- Open split mode. In one pane, run queries trough backend, in second run it trough proxy (exemplars toggled on). The results should be the same.

**Test queries with Both query type selected:**
1. `rate(go_goroutines[$__rate_interval])` with 24h step and 7 day interval. The range queries should be aligned 00:00:00 each day and instant query should have current time stamp.
